src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -506,6 +528,24 @@ bool CartGridStrategy::doProbe(Gcode *gc)
         return false;
     }
 
+    if (do_manual_attach) {
+        // Move to the attachment point defined
+        if (do_home) zprobe->home();
+
+        float x, y, z;
+        std::tie(x, y, z) = m_attach;
+        zprobe->coordinated_move( x, y, z , zprobe->getFastFeedrate());
+
+        gc->stream->printf(" ************************************************************\n");
+        gc->stream->printf("     Ensure probe is attached and trigger probe when done\n");
+        gc->stream->printf(" ************************************************************\n");
+
+        while( !zprobe->getProbeStatus()) {
+            if(THEKERNEL->is_halted()) break;
```

Comment:
Actually `return false;` would be more appropriate than `break` if you are not going to test when it exists the loop. Basically this would abort the probe if control X is hit (or the kill button).